diff --git a/node_modules/@libp2p/mplex/dist/src/index.d.ts b/node_modules/@libp2p/mplex/dist/src/index.d.ts
index 14c5f6e..32d0ee0 100644
--- a/node_modules/@libp2p/mplex/dist/src/index.d.ts
+++ b/node_modules/@libp2p/mplex/dist/src/index.d.ts
@@ -1,6 +1,7 @@
 import { Components, Initializable } from '@libp2p/components';
 import type { StreamMuxerFactory, StreamMuxerInit } from '@libp2p/interface-stream-muxer';
 import { MplexStreamMuxer } from './mplex.js';
+export { MplexStreamMuxer } from './mplex.js';
 export interface MplexInit {
     /**
      * The maximum size of message that can be sent in one go in bytes.
diff --git a/node_modules/@libp2p/mplex/dist/src/index.js b/node_modules/@libp2p/mplex/dist/src/index.js
index 4d692df..7bf592c 100644
--- a/node_modules/@libp2p/mplex/dist/src/index.js
+++ b/node_modules/@libp2p/mplex/dist/src/index.js
@@ -1,5 +1,6 @@
 import { Components } from '@libp2p/components';
 import { MplexStreamMuxer } from './mplex.js';
+export { MplexStreamMuxer } from './mplex.js';
 export class Mplex {
     constructor(init = {}) {
         this.protocol = '/mplex/6.7.0';
